packages/swagger-ts/src/TypeGenerator.ts

```diff
@@ -386,8 +400,47 @@ export class TypeGenerator extends Generator<PluginOptions['resolvedOptions'], C
         })
       }
 
-      if (this.options.dateType === 'date' && ['date', 'date-time'].some((item) => item === schema.format)) {
-        return factory.createTypeReferenceNode(factory.createIdentifier('Date'))
+      /**
+       * > Structural validation alone may be insufficient to allow an application to correctly utilize certain values. The "format"
+       * > annotation keyword is defined to allow schema authors to convey semantic information for a fixed subset of values which are
+       * > accurately described by authoritative resources, be they RFCs or other external specifications.
+       *
+       * In other words: format is more specific than type alone, hence it should override the type value, if possible.
+       *
+       * see also https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-validation-00#rfc.section.7
+       */
+      if (schema.format) {
+        switch (schema.format) {
+          case 'binary':
+            if (schema.type === 'string') {
+              return factory.createTypeReferenceNode('Blob', [])
+            }
+            break
+          // 7.3.1. Dates, Times, and Duration
+          case 'date-time':
+          case 'date':
+          case 'time':
+            if (this.options.dateType === 'date') {
+              return factory.createTypeReferenceNode(factory.createIdentifier('Date'))
+            }
+          case 'uuid':
+            // TODO how to work with this? use https://www.npmjs.com/package/uuid for it?
```

Comment:
Currently, `uuid` would be ignored like all the others, although it would made sense to make it an `uuid`, but this would require a custom task or a dependency `uuid`

Reply:
In the end, it will be a `string`, Typescript does not have a type uuid so I don't think we need to change something here for the ts plugin.

Reply:
The thing is, the cases added here are directly from the openapi spec. That's why I added them here, too. 
But I'll remove unsupported cases if you like. 

Imo, being able to map to a specific type is very convinient. But kubb itself should add as little runtime dependencies as possible. Hence, uuid support via `@types/uuid` and `uuid` would be a great use case for a kubb plugin? Is that already possible with plugins, then I might give them a try.